smart-contract-usage-in-javascript-sdk.md

```diff
@@ -0,0 +1,326 @@
+
+    // results from the node need to be decoded to be human readable
+    const decodedError = await client.contractDecodeData('string', error.returnValue).catch(logError);
+    console.error('error:', decodedError.value);
+};
+```
+
+## Define Example Contract
+
+We define an example contract which features functions that can reserve something if a minimum price is payed to the contract. The contract creator can later withdraw the payed tokens. Each line is commented with what it does. Sophia language defintion can be found in [the protocol repository](https://github.com/aeternity/protocol/blob/master/contracts/sophia.md).
+
+Save this as `ExampleContract.aes` in the same directory as `index.js`.
+
+```
+// contract definition
+contract ExampleContract =
+
+
+  // state definiton, with available count, set minimum price and the owner
+  record state = { available_count : int, min_price : int, owner : address }
+
+
+  // defining the require function later used to check calling constraints
+  private function require(b : bool, err : string) =
+    if(!b) abort(err)
+
+
+  // function to initialize the contract, setting the available count, minimum price and the owner
+  public function init(available_count' : int, min_price' : int) : state =
+    { available_count = available_count', min_price = min_price', owner = Call.caller }
+
+
+  // function for anyone to reserve some count requiring available count and the minimum price payed
+  public stateful function reserve(count: int) =
+  
+    // requiring the price payed per item to be above or equal the set minimum price
+    require(Call.value >= state.min_price * count, "Payed price is less than minimum required")
+    let remaining_count = state.available_count - count
+    
+    // requiring count to be available
+    require(remaining_count >= 0, "Less available then to be reserved")
+    
+    // applying remaining count to the state
+    put(state{ available_count = remaining_count })
+
+
+  // function to get the remaining count, can be statically called as it doesn't change state
+  public function available_count() : int =
+    state.available_count
+
+
+  // function only to be called by the owner to withdraw all balance that was sent to the contract
+  public stateful function owner_withdraw() : int=
+  
+    // requiring contract caller to be the registered owner
+    require(Call.caller == state.owner, "Only owner can withdraw")
+    let withdrawed_balance = Contract.balance
+    
+    // creating spend of the balance to the registered owner
+    Chain.spend(state.owner, Contract.balance)
+    
+    // returning the balance that was withdrawed
+    withdrawed_balance
+```
+
+## SDK Contracts Workflow
+
+
+### 1. Initialize SDK client
+
+Inside the main function, to create the SDK client referencing the æternity network node and the needed public and private keys, add after the previously added lines:
+
+```
+// initialize aeternity sdk client
```

Comment:
In order to avoid confusion you can add 
```
const main = async (name) => {
...
The code here
...
};
```
This way the reader won't place this at the bottom of the index.js file